reference/win32service/functions/win32-add-right-access-service.xml

```
@@ -0,0 +1,145 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- $Revision$ -->
+<!-- Generated by xml_proto.php v2.3. Found in /scripts directory of phpdoc. -->
+<refentry xml:id="function.win32-add-right-access-service" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
+ <refnamediv>
+  <refname>win32_add_right_access_service</refname>
+  <refpurpose>Add rights access for an username to the service </refpurpose>
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <type>void</type><methodname>win32_add_right_access_service</methodname>
+   <methodparam><type>string</type><parameter>servicename</parameter></methodparam>
+   <methodparam><type>string</type><parameter>username</parameter></methodparam>
+   <methodparam><type>int</type><parameter>right</parameter></methodparam>
+   <methodparam choice="opt"><type>string</type><parameter>machine</parameter><initializer>&null;</initializer></methodparam>
+  </methodsynopsis>
```

Not directly related to this PR, but consider to introduce stub.php files in the codebase. These can be used to generate the arginfo of the functions, and also to generate the methodsynopses for the documentation (both using build/gen_stub.php from php-src). This way there is a single source of truth, and it's nicely readable as well (even for non C programmers).

---

Have you a example?

---

Well, you can look into php-src (I believe that completely uses stubs now), or maybe see https://github.com/php/pecl-database-dbase/commit/c12a7c35f4373791e6605bd1da8840c514fc9b4a (the changes  regarding throwing instead of returning false are not directly related; you can ignore these): dbase.stub.php is hand-written, and dbase_arginfo.h has been automatically generated. dbase_arginfo.h is then included in dbase.c where the previous arginfo had been removed. Or see https://github.com/php/php-src/pull/4284, where this started.

---

It looks good. Do I need to get the arginfo generator script from the php-src repository?

---

You can get it from there, or from a devel package from windows.php.net (it's included there for PHP 8.2, and maybe earlier).